keel-docker/src/test/kotlin/com/netflix/spinnaker/keel/docker/SampleDockerImageResolver.kt

```diff
@@ -22,14 +22,18 @@ import com.netflix.spinnaker.keel.api.ApiVersion
 import com.netflix.spinnaker.keel.api.Resource
 import com.netflix.spinnaker.keel.api.ResourceSpec
 import com.netflix.spinnaker.keel.api.artifacts.DockerArtifact
+import com.netflix.spinnaker.keel.api.plugins.SupportedKind
 import com.netflix.spinnaker.keel.persistence.KeelRepository
 
 class SampleDockerImageResolver(
   repository: KeelRepository
 ) : DockerImageResolver<SampleSpecWithContainer>(
   repository
 ) {
-  override val supportedKind = SAMPLE_API_VERSION.qualify("sample")
+  override val supportedKind = SupportedKind(
```

Comment:
I've refactored `Resolver` implementations to use `SupportedKind`, similar to what was done earlier for `ResourceHandlers`, so that we could use the `specClass` field to filter resolvers for those that support implementations of `ComputeResourceSpec`.